debezium-core/src/main/java/io/debezium/relational/HistorizedRelationalDatabaseSchema.java

```diff
@@ -41,12 +49,27 @@ protected HistorizedRelationalDatabaseSchema(HistorizedRelationalDatabaseConnect
     }
 
     @Override
-    public void recover(OffsetContext offset) {
+    public void recover(TaskOffsetContext<P, O> taskOffsetContext) {
+        final Map<P, O> taskOffsets = taskOffsetContext.getOffsets();
+        final List<O> nonEmptyTaskOffsets = taskOffsets.values().stream()
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+
+        // exit early as there is nothing to recover
+        if (nonEmptyTaskOffsets.isEmpty()) {
+            return;
+        }
```

Comment:
This could be simplified to:
```java
final boolean hasNonNullOffsets = offsets.getOffsets()
    .values()
    .stream()
    .anyMatch(Objects::nonNull);

if (!hasNonNullOffsets) {
    // there is nothing to recover
    return;
}
```
This way, the stream will abort as soon as there's a match w/o having to collect all matches.